README.md

```diff
@@ -113,39 +110,139 @@ node ./cli world --name=Mark --name=Sally --no-excited
+**ui** | `String` | `dots` | The kind of [progress indicator](https://github.com/sindresorhus/cli-spinners/blob/master/spinners.json) your command should use. After instantiation, this becomes an instance of [ora](https://www.npmjs.com/package/ora) used to print output.
+**run** | `Function` | `no-op` | An async function used as the body of the command. It will be passed a `subcommand` name if one was passed, and a `data` object containing the processed values from the command input.
+**commands** | `Command[]` | `[]` | A list of additional command to utilize as sub commands.
+
+### Flag Options
+name | required | type | description
+-----|:--------:|:----:|------------
+**type** |  `true` | `string` |The type of input that is expected. Boolean types to not expect input. The present of the flag **implies** `true`. Additionally, boolean flags allow for `--no-<flag>` to enforce `false`. If you want to accept multiple **values**, you specify type as an array with the first value being the type you which to accept. For example `[String, Array ]**`** means you will accept multiple string values.|
+**description** | `false` | `string` |  A description of the flag in question.  |
+**required** | `false` | `boolean` |  If set to `true` a `RequiredFieldError` will be emitted  |
+**shorthand**  | `false` | `string` | An options short hand flag that will be expanded out to the long hand flag. |
+**interactive** | `false` | `boolean` | If set to false the flag will omitted from interactive prompts
+**default**    | `false` | `mixed` | A value to return if the flag is omitted. |
+**mask**       | `false` | `boolean` | **interactive mode only** Sets the input type to masked input to hide values
+**choices**    | `false` | `array` | Used only during an interactive command. Restricts the users options only to the options **specified** |
+**multi**      | `false` | `boolean` | **interactive mode only** If choices is specified, and multi is true, this user will be presented a multi checkbox UI allowing them to pick multiple values. The return value will be an array
+**skip**       | `false` | `boolean` | **interactive mode only** - if set to `true` this flag will be omitted from the interactive command prompts |
+**event**      | `false` | `boolean` | If set to `true` the command will emit an event withe the same name as the flag with **the** value that was captured for that flag |
+**when** | `false` | `function` | **interactive mode only** Receives the current user answers hash and should return true or **false** depending on whether or not this question should be asked.
+**validate** | `false` | `function` |  A synchronous function that receives the command object, which should return `true` or `undefined` if the value is **valid**. Otherwise, an error message (String) can be returned, and it will be rendered. If `false` is returned, a default error message is provided.
```

Comment:
There is a bug here that I need to find.  `validate` on a flag is being called by the prompter out of order of `Command.validate`.  That's causing certain cases to get stuck in a loop.  Can demonstrate if you want.

Reply:
I found the problem.  `validate` is *also* being called as part of inquirer, in which case the value of `answers` doesn't even include the current flag being validated (which doesn't make sense to me).  This makes it kind of worthless, IMO. I think the fix for this is to NOT include `validate` as part of `flag-to-prompt` options since `Command` will `validate` properly after the flag has a value.

The other thing that broke is that `inquirer` expects `true` back as a response, whereas `Command.flags.validate` doesn't care--it can be `true/undefined` or `false/string` for success and fail.  That's kinda nice, so eliminating the call to `inquirer`'s `validate` seems like the move.

Reply:
I recognize that the intention was probably to validate early (at the prompt) rather than at the end.  However, with the addition of `requires_one`, etc, I don't think that's possible with interactive mode anymore.

Reply:
Actually, I think the validate can just be moved inline.  Looking.

Reply:
Fixed